site/src/components/mdx/mdx-page-content.tsx

```diff
@@ -143,12 +156,14 @@ export const MdxPageContent: FunctionComponent<MdxPageContentProps> = ({
       }
     };
 
-    window.addEventListener("scroll", onScroll);
+    const throttledOnScroll = throttle(onScroll, 1000);
+
+    window.addEventListener("scroll", throttledOnScroll);
 
     return () => {
-      window.removeEventListener("scroll", onScroll);
+      window.removeEventListener("scroll", throttledOnScroll);
     };
-  }, [router, headings, detectHeadingFromScroll]);
```

Comment:
switched these to refs, since we don't need this effect to run if any of them changes